### PR TITLE
Improve Task Kill behavior in deployments by introducing kills in batches

### PR DESCRIPTION
**Pull request by aquamatthias**
The limit and cycle can be controlled via internal configuration parameters.
Introduces a config, that is passed to the deployment actors.
Adapted tests to meet the new configuration parameter.
Adapted tests to align with the new logic.
Additional tests added for the new functionality.

### CODE REVIEW

**Comment by gkleiman** [comment]
@aquamatthias `ResidentTaskIntegrationTest` failed, did this change break it or is it flaky?

**Comment by aquamatthias** [comment]
It build on Wednesday - let me look.

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -158,7 +159,8 @@ class MarathonSchedulerActor private (
       val origSender = sender()
       withLockFor(appId) {
         val promise = Promise[Unit]()
-        context.actorOf(TaskKillActor.props(driver, appId, taskTracker, eventBus, taskIds, promise))
+        context.actorOf(
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Doesn't this line fit in 120 columns?

File: src/main/scala/mesosphere/marathon/upgrade/AppStopActor.scala [hunk 2]
```diff
@@ -15,14 +15,21 @@ class AppStopActor(
     val taskTracker: TaskTracker,
     val eventBus: EventStream,
     app: AppDefinition,
+    val config: UpgradeConfig,
     val promise: Promise[Unit]) extends StoppingBehavior {
 
   override var idsToKill: mutable.Set[Task.Id] = taskTracker.appTasksLaunchedSync(app.id).map(_.taskId).to[mutable.Set]
 
   def appId: PathId = app.id
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/upgrade/AppStopActor.scala]: I know you didn't add this line, but could you also add `override` to the method declaration?

File: src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+  * Config parameters for the upgrade module.
+  */
+trait UpgradeConfig extends ScallopConf {
+  //scalastyle:off magic.number
+
+  lazy val killBatchSize = opt[Int]("killBatchSize",
+    descr = "INTERNAL TUNING PARAMETER: " +
+      "The number of kills that can be issued in one batch size.",
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala]: does `killsPerBatch` sound better?

**Comment by gkleiman** [comment]
Just 2 very minor comments, LGTM!

File: src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala (hunk 3 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala]: Yes!

File: src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  lazy val killBatchDuration = opt[Long]("killBatchDuration",
+    descr = "INTERNAL TUNING PARAMETER: " +
+      "The duration of one batch cycle in milliseconds.",
+    noshort = true,
+    hidden = true,
+    default = Some(10000L) //10 seconds
+  )
+
+  def batchCycle: FiniteDuration = killBatchDuration().millis
+  def batchSize: Int = killsPerBatch()
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala]: Do we really need this method?

File: src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    */
+  lazy val killBatchDuration = opt[Long]("killBatchDuration",
+    descr = "INTERNAL TUNING PARAMETER: " +
+      "The duration of one batch cycle in milliseconds.",
+    noshort = true,
+    hidden = true,
+    default = Some(10000L) //10 seconds
+  )
+
+  def batchCycle: FiniteDuration = killBatchDuration().millis
```
Comment (gkleiman) [inline src/main/scala/mesosphere/marathon/upgrade/UpgradeConfig.scala]: This returns a duration, so to be consistent with other config classes (such as `ZookeeperConf`), I'd call the method `killBatchDuration` and the rename the variable to something like `killBatchTimeout`,  `killBatchPeriod` or  `killBatchLength`.

**Comment by gkleiman** [comment]
:ship:

**Pull request merged**